client.go

```diff
@@ -659,9 +669,18 @@ type DialFuncWithTimeout func(addr string, timeout time.Duration) (net.Conn, err
 // Request argument passed to RetryIfFunc, if there are any request errors.
 type RetryIfFunc func(request *Request) bool
 
-// RetryIfErrFunc defines the signature of the retry if error function.
-// request and error passed to RetryIfErrFunc, if there are any request errors.
-type RetryIfErrFunc func(request *Request, err error) bool
+// RetryIfErrFunc defines an interface used for implementing the following functionality:
+// When the client encounters an error during a request, the behavior—whether to retry,
+// whether to retry later, or whether to reset the request timeout—should be determined
+// based on the return value of this interface.
+//
+// retryCount indicates which retry attempt this is, with retry counting starting from 1.
+// The first request execution is not considered a retry.
+// err represents the error encountered during the previous request.
+// sleepDuration if is greater than 0, the retry will be executed after this interval.
+// resetReqTime indicates whether to reset the request timeout to its initial value before executing this retry.
+// retry indicates whether to perform this retry. If false, the request function will return immediately.
+type RetryIfErrFunc func(request *Request, retryCount int, err error) (sleepDuration time.Duration, resetReqTime bool, retry bool)
```

Comment:
What is the reason for returning the `sleepDuration`? The user of this function can more easily just call `time.Sleep()` in the `RetryIfErrFunc` itself.